### PR TITLE
chore: auth task logs [DET-7554]

### PR DESCRIPTION
**Pull request by NicholasBlaskey**
## Description

Adds a new GRPC endpoint.

Removes old task logs. I think we should just remove it now since it will be a breaking change adding auth to it anyway.

## Test Plan

Integration test + e2e tests cover task logs pretty well

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

**Review by stoksc — APPROVED** [review]
lgtm, but @rb-determined-ai needs to see this

**Comment by stoksc** [comment]
(or you/someone on backend needs to update the new script once he lands it)

File: master/internal/api_tasks.go [hunk 1]
```diff
@@ -300,6 +300,43 @@ func (a *apiServer) TaskLogs(
...
+	taskID := req.Logs[0].TaskId
+
+	if err := a.canDoActionsOnTask(ctx, model.TaskID(taskID),
+		expauth.AuthZProvider.Get().CanEditExperiment); err != nil {
+		return nil, err
+	}
+
+	logs := make([]*model.TaskLog, len(req.Logs))
+	for i := range req.Logs {
+		if req.Logs[i].Id != nil {
```
Comment (corban-beaird) [inline master/internal/api_tasks.go:318]: question (non-blocking): Why are we checking this field? And why is it required to be nil? Is it prevent confusion?

File: master/internal/api_tasks.go [hunk 2]
```diff
@@ -300,6 +300,43 @@ func (a *apiServer) TaskLogs(
...
+			return nil, status.Errorf(codes.InvalidArgument,
+				"ID must be nil on logs got %d instead", *req.Logs[i].Id)
+		}
+		if req.Logs[i].TaskId != taskID {
+			// There isn't a hard reason for this requirement other than we would have to RBAC
+			// against all provided taskIDs. This usecase seems pretty unlikely.
```
Comment (corban-beaird) [inline master/internal/api_tasks.go:324]: question (non-blocking): Do you know of any cases where this could occur? Or is this preventive and/or for checking assumptions?

File: master/internal/api_tasks_intg_test.go [hunk 3]
```diff
@@ -6,19 +6,95 @@ package internal
...
 	authz2 "github.com/determined-ai/determined/master/internal/authz"
 	"github.com/determined-ai/determined/master/pkg/model"
+	"github.com/determined-ai/determined/master/pkg/ptrs"
 	"github.com/determined-ai/determined/proto/pkg/apiv1"
 	"github.com/determined-ai/determined/proto/pkg/checkpointv1"
+	"github.com/determined-ai/determined/proto/pkg/logv1"
+	"github.com/determined-ai/determined/proto/pkg/taskv1"
 )
 
+func TestPostTaskLogs(t *testing.T) {
```
Comment (corban-beaird) [inline master/internal/api_tasks_intg_test.go:28]: I like this test a lot, it's clear and effectively covers edge cases

**Review by corban-beaird — APPROVED** [review]
Looks great!

File: master/internal/api_tasks.go (hunk 1 above)
Comment (NicholasBlaskey) [reply master/internal/api_tasks.go:318]: We ignore this field in insert task logs since it is autogenerated at the db.

https://github.com/determined-ai/determined/blob/588d817d70bbfdc4860c6d1262be061cdce9761c/master/internal/db/postgres_tasks.go#L362C1-L362C22

I just made it required to be nil so people aren't setting something then wondering why it has no effect.

I guess the field is technically unused but we should probably switch some endpoints to use the task logs proto object for consistency eventually

File: master/internal/api_tasks.go (hunk 2 above)
Comment (NicholasBlaskey) [reply master/internal/api_tasks.go:324]: Not from our ```enrich_task_logs.py```, but you could technically just call this endpoint from user code and pass in whatever arguments you want causing something like this to happen.

**Review by rb-determined-ai — APPROVED** [review]
I'll just tweak my own log shipper after this lands... I'm pretty sure mine will land second.

**Review by rb-determined-ai — COMMENTED** [review]
Though I do _not_ agree with removing echo endpoints in this PR.

If leaving it there for 3 months means we have 90% fewer support tickets about this, that's a win in my book.
